vars/mtaBuild.groovy

```diff
@@ -65,8 +67,20 @@ def call(Map parameters = [:]) {
             }
         }
 
-        def mtaYaml = readYaml file: "${pwd()}/mta.yaml"
+        def mtaYmlName = "${pwd()}/mta.yaml"
+        def applicationName = configuration.applicationName
+
+        if (!fileExists(mtaYmlName)) {
+            if (!applicationName) {
+                echo "'applicationName' not provided as parameter - will not try to generate mta.yml file"
```

Comment:
Is it possible to continue here? We do not have a mtaYaml file and we do not have an application name, which is needed for generating such a file. Finally we will not have that file. Is is possible to do anything reasonable without this file. I don't know ... In case it is not possible to continue without that file it would be better IMO to fail here (fail fast). @IKirilova : in case it is reasonable to continue here, please let me know. In this case we can merge, I suppose.

Reply:
As discussed with IKirilova: In case the file does not exist we fail with a FileFoundException and we have the log message prior to that. We failed also before that change with a file not found exception triggered by readYaml. This behavour is the same now.